### PR TITLE
docs: update `CONTRIBUTING.md`, PR template

### PR DESCRIPTION
**Pull request by Bertie690**
## What are the changes the user will see?
N/A
## Why am I making these changes?
I thought the section could use a bit of updating, not to mention some info was all over the place.

## What are the changes from a developer perspective?
- Moved some explanatory comments into their relevant sections
- Made the checklist part more explicitly grouped
- Added sub-headers to `CONTRIBUTING.md` while I had the opportunity (and added links to Vitest's docs)

## Screenshots/Videos
N/A
## How to test the changes?
N/A

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR
- [x] The PR is self-contained and cannot be split into smaller PRs
- [x] I have provided a clear explanation of the changes
- [x] The PR title matches the format described in [CONTRIBUTING.md](https://github.com/pagefaultgames/pokerogue/blob/beta/CONTRIBUTING.md#-submitting-a-pull-request)
- [x] I have tested the changes manually

### CODE REVIEW

**Review by Madmadness65 — APPROVED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -17,43 +20,38 @@ fix(move): Future Sight no longer crashes
...
-The first section is mandatory if there are user-facing changes (it will be used as the base for a changelog entry).
-The second and third section are mandatory.
-The screenshot/video section is mandatory if you made any visual changes (such as to UI elements).
--->
+Summarize the changes from a user perspective on the application.
+Try to keep this section (relatively) brief as it is used to generate changelogs.
 
-## What are the changes the user will see?
-<!-- Summarize what are the changes from a user perspective on the application -->
+PRs with no user-facing changes should instead write "N/A" or similar to be ommitted from auto-generated changelogs.
```
Comment (DayKev) [inline .github/pull_request_template.md]:
```suggestion
PRs with no user-facing changes should instead write "N/A" to be ommitted from auto-generated changelogs.
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -77,26 +75,31 @@ Do the reviewers need to do something special in order to test your changes?
...
 - [ ] I have provided screenshots/videos of the changes (if applicable)
-  - [ ] I have made sure that any UI change works for both the default and legacy UI themes (if applicable)
+  - [ ] I have made sure that any UI changes work for both the default and legacy UI themes (if applicable)
+- [ ] **I have notified the Dev Team about this PR's existence** (such as by posting inside the current PR callout thread on Discord)
 
 Are there any localization additions or changes? If so:
-- [ ] A locales PR been created on the [locales](https://github.com/pagefaultgames/pokerogue-locales) repo
-  - [ ] Link to locales PR: 
-- [ ] The translation team been contacted for proofreading/translation on Discord
+- [ ] I have created an associated PR on the [locales](https://github.com/pagefaultgames/pokerogue-locales) repository (pagefaultgames/pokerogue-locales#<PR_NUMBER_HERE>)
```
Comment (DayKev) [inline .github/pull_request_template.md]: Rewriting it this way (instead of having the extra bullet point for linking the PR) introduces ambiguity, which should be avoided.

Same with the assets one below.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -133,18 +133,28 @@ You can get help testing your specific changes, and you might have found a new o
...
-    - Are deterministic. In other words, the test should never pass or fail when it shouldn't due to randomness. This involves primarily ensuring that abilities and moves are never randomly selected.
-    - As much as possible, are unit tests. If you have made two distinct changes, they should be tested in two separate cases.
-    - Test edge cases. A good strategy is to think of edge cases beforehand and create tests for them using `it.todo`. Once the edge case has been handled, you can remove the `todo` marker.
 
+To create a new test file, run `pnpm test:create` and follow the prompts. \
+If the move/ability/etc. you're modifying already has tests, you can add new cases to the test file or edit existing ones.
+
+- Ensure that new test cases:
+  - Are deterministic. In other words, the test should never pass or fail when it shouldn't due to randomness. Among other things, this involves ensuring that abilities and moves are never randomly selected.
+  - Are unit tests. If you have made two distinct changes, they should be tested in two separate cases.
```
Comment (DayKev) [inline CONTRIBUTING.md]:
```suggestion
  - Do not test multiple separate things in the same test case. If you have made two distinct changes, they should be tested in two separate cases.
```

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -77,26 +75,31 @@ Do the reviewers need to do something special in order to test your changes?
...
+Please ensure the following requirements are all met before creating your PR.
+If this is not the case, consider [marking the PR as a draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) until all bullets have been resolved.
+
+If an item or category isn't valid for the particular changes being made (for example, you didn't make any locales changes)
+you can strike it out with the `~` character to mark them as not applicable.
 -->
-- [ ] **I'm using `beta` as my base branch**
-- [ ] There is no overlap with another PR
+
+- [ ] The PR is correctly formatted
```
Comment (DayKev) [inline .github/pull_request_template.md]:
```suggestion
- The PR is correctly formatted:
```

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -77,26 +75,31 @@ Do the reviewers need to do something special in order to test your changes?
...
+  - [ ] I have provided a clear explanation of the changes within the PR description
+  - [ ] The PR title matches the Conventional Commits format (as described in [CONTRIBUTING.md](../CONTRIBUTING.md#pr-title-format))
 - [ ] The PR is self-contained and cannot be split into smaller PRs
-- [ ] I have provided a clear explanation of the changes
-- [ ] The PR title matches the format described in [CONTRIBUTING.md](https://github.com/pagefaultgames/pokerogue/blob/beta/CONTRIBUTING.md#-submitting-a-pull-request)
-- [ ] I have tested the changes manually
-- [ ] The full test suite still passes (`pnpm test:silent`)
-  - [ ] I have created new automated tests (`pnpm test:create`) or updated existing tests related to the PR's changes
+  - [ ] There is no overlap with another open PR
+- [ ] The PR has been confirmed to work correctly
```
Comment (DayKev) [inline .github/pull_request_template.md]:
```suggestion
- The PR has been confirmed to work correctly:
```

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -17,43 +20,38 @@ fix(move): Future Sight no longer crashes
...
 <!--
 Explain why you decided to introduce these changes.
-Does it come from an issue or another PR? Link to them if possible.
-Explain why you believe this can enhance user experience.
--->
-<!--
-If there are existing GitHub issues related to the PR that would be fixed,
-you can add "Fixes #[issue number]" (e.g.: "Fixes #1234") to link an issue to your PR
+Does it come from another issue, PR or other prior discussion? Link to them if possible.
+Explain why you believe this can enhance user experience or otherwise improve the codebase.
```
Comment (Wlowscha) [inline .github/pull_request_template.md]:
```suggestion
Explain why you believe this can enhance user experience or otherwise improve the codebase.
Keep this explanation objective as much as possible, and avoid referring to personal feelings, or making derogatory comments about existing code.
```

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -1,10 +1,13 @@
-<!-- Once you have read these comments, you are free to remove them -->
+<!--
+Thank you for contributing to PokéRogue!
+Open-source contributions like yours are what keep this project going.
```
Comment (SirzBenjie) [inline .github/pull_request_template.md]:
```suggestion
Open-source contributions like yours help keep this project going.
```

File: .github/pull_request_template.md [hunk 8]
```diff
@@ -1,10 +1,13 @@
...
 
-<!-- Feel free to look at other PRs for examples -->
+Note that these comment blocks are purely informative and can be removed once you're done reading them.
+-->
 
 <!--
 Make sure your title matches the https://www.conventionalcommits.org/en/v1.0.0/ format.
-Ideally the title is less than or equal to 72 characters (GitHub cuts off commit titles longer than this length).
+Attempt to make the title no longer than 72 characters (GitHub cuts off commit titles longer than this length).
```
Comment (SirzBenjie) [inline .github/pull_request_template.md]:
```suggestion
Try to make the title no longer than 72 characters, as GitHub cuts off commit titles longer than this length.
```

File: .github/pull_request_template.md [hunk 9]
```diff
@@ -17,43 +20,41 @@ fix(move): Future Sight no longer crashes
...
-The first section is mandatory if there are user-facing changes (it will be used as the base for a changelog entry).
-The second and third section are mandatory.
-The screenshot/video section is mandatory if you made any visual changes (such as to UI elements).
--->
+Summarize the changes from a user perspective on the application.
+Try to keep this section (relatively) brief as it is used to generate changelogs.
 
-## What are the changes the user will see?
-<!-- Summarize what are the changes from a user perspective on the application -->
+PRs with no user-facing changes should instead write "N/A" to be ommitted from auto-generated changelogs.
```
Comment (SirzBenjie) [inline .github/pull_request_template.md]:
```suggestion
PRs with no user-facing changes should leave this blank or write "N/A" to be omitted from auto-generated changelogs.
```

File: .github/pull_request_template.md [hunk 10]
```diff
@@ -17,43 +20,41 @@ fix(move): Future Sight no longer crashes
...
-<!--
-If there are existing GitHub issues related to the PR that would be fixed,
-you can add "Fixes #[issue number]" (e.g.: "Fixes #1234") to link an issue to your PR
+Does it come from another issue, PR or other prior discussion? Link to them if possible.
+How can this can enhance user experience or otherwise improve the codebase?
+
+Try to keep this explanation as objective as possible — avoid referring to personal feelings or making derogatory comments about existing code.
+
+If this PR fixes an existing GitHub issue,
+you can add "Fixes #[issue number]" (e.g.: "Fixes #1234") to link the issue
```
Comment (SirzBenjie) [inline .github/pull_request_template.md]:
```suggestion
If this PR resolves an existing GitHub issue,
you can add "Fixes #[issue number]" (e.g.: "Fixes #1234") to link the issue. (See https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue)
```

File: .github/pull_request_template.md [hunk 11]
```diff
@@ -77,26 +78,33 @@ Do the reviewers need to do something special in order to test your changes?
 
 ## Checklist
 <!--
-If an item isn't valid (for example, you didn't make any locales changes)
-you can cross it out using the tilde character (~), e.g.:
-- ~[ ] A locales PR been created on the [locales](https://github.com/pagefaultgames/pokerogue-locales) repo~
+Please ensure the following requirements are all met before creating your PR.
+If this is not the case, consider [marking the PR as a draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) until all bullets have been resolved.
```
Comment (SirzBenjie) [inline .github/pull_request_template.md]:
```suggestion
If this is not the case, consider marking the PR as a draft (https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) until all bullets have been resolved.
```

The markdown comments won't format as actual link replacements, so no need to use that syntax ;).

File: .github/pull_request_template.md [hunk 12]
```diff
@@ -77,26 +78,33 @@ Do the reviewers need to do something special in order to test your changes?
...
+
+If an item or category isn't valid for the particular changes being made (for example, you didn't make any locales changes)
+you can strike it out with the `~` character to mark them as not applicable.
 -->
-- [ ] **I'm using `beta` as my base branch**
-- [ ] There is no overlap with another PR
+
+- The PR is correctly formatted:
```
Comment (SirzBenjie) [inline .github/pull_request_template.md]:
```suggestion
- The PR is correctly formatted:
```
```suggestion
- The PR content is correctly formatted:
```

File: .github/pull_request_template.md [hunk 13]
```diff
@@ -77,26 +78,33 @@ Do the reviewers need to do something special in order to test your changes?
...
 - [ ] The PR is self-contained and cannot be split into smaller PRs
-- [ ] I have provided a clear explanation of the changes
-- [ ] The PR title matches the format described in [CONTRIBUTING.md](https://github.com/pagefaultgames/pokerogue/blob/beta/CONTRIBUTING.md#-submitting-a-pull-request)
-- [ ] I have tested the changes manually
-- [ ] The full test suite still passes (`pnpm test:silent`)
-  - [ ] I have created new automated tests (`pnpm test:create`) or updated existing tests related to the PR's changes
+  - [ ] There is no overlap with another open PR
+- The PR has been confirmed to work correctly:
+  - [ ] I have tested the changes manually
+  - [ ] The full automated test suite still passes
```
Comment (SirzBenjie) [inline .github/pull_request_template.md]:
```suggestion
  - [ ] The full automated test suite still passes (use `pnpm test:silent` to test locally)
```

File: .github/pull_request_template.md [hunk 14]
```diff
@@ -77,26 +78,33 @@ Do the reviewers need to do something special in order to test your changes?
...
+  - [ ] I have created new automated tests (`pnpm test:create`) or updated existing tests related to the PR's changes if necessary
 - [ ] I have provided screenshots/videos of the changes (if applicable)
-  - [ ] I have made sure that any UI change works for both the default and legacy UI themes (if applicable)
+  - [ ] I have made sure that any UI changes work for both the default and legacy UI themes (if applicable)
+- [ ] **I have notified the Dev Team about this PR's existence** (such as by posting inside the current PR callout thread on Discord)
```
Comment (SirzBenjie) [inline .github/pull_request_template.md]:
```suggestion
```

I don't think we need this. We regularly check PRs. @Madmadness65 is always on top of it, too.

File: .github/pull_request_template.md [hunk 15]
```diff
@@ -77,26 +78,33 @@ Do the reviewers need to do something special in order to test your changes?
...
 
 Are there any localization additions or changes? If so:
-- [ ] A locales PR been created on the [locales](https://github.com/pagefaultgames/pokerogue-locales) repo
-  - [ ] Link to locales PR: 
-- [ ] The translation team been contacted for proofreading/translation on Discord
+- [ ] I have created an associated PR on the [locales](https://github.com/pagefaultgames/pokerogue-locales) repository
+  - If so, include a link to the PR here: _____
```
Comment (SirzBenjie) [inline .github/pull_request_template.md:104]:
```suggestion
  - If so, include a link to the PR here: 
```

File: .github/pull_request_template.md [hunk 16]
```diff
@@ -77,26 +78,33 @@ Do the reviewers need to do something special in order to test your changes?
...
+- [ ] I have contacted the Translation Team on Discord for proofreading/translation
 
-Does this require any changes to the assets folder? If so:
-- [ ] A PR been created on the [assets](https://github.com/pagefaultgames/pokerogue-assets) repo
-  - [ ] Link to assets PR: 
\ No newline at end of file
+Does this require any additions or changes to in-game assets? If so:
+- [ ] I have created an associated PR on the [assets](https://github.com/pagefaultgames/pokerogue-assets) repository
+  - If so, include a link to the PR here: _____
```
Comment (SirzBenjie) [inline .github/pull_request_template.md:109]:
```suggestion
  - If so, include a link to the PR here: 
```

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -133,18 +132,30 @@ You can get help testing your specific changes, and you might have found a new o
 
 ### 2 - Automatic Testing
 
-> PokéRogue uses [Vitest](https://vitest.dev/) for automatic testing. Checking out the existing tests in the [test](./test/) folder is a great way to understand how this works, and to get familiar with the project as a whole.
+PokéRogue uses [Vitest](https://vitest.dev/) for automated testing. \
```
Comment (SirzBenjie) [inline CONTRIBUTING.md]:
```suggestion
PokéRogue uses [Vitest](https://vitest.dev/) for automated testing.

```

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -133,18 +132,30 @@ You can get help testing your specific changes, and you might have found a new o
...
+Checking out existing tests in the [test](./test/) folder is a great way to understand how the existing system works, as well as familiarizing yourself with the project as a whole.
+
+#### Writing tests
+Most non-trivial changes (_especially bug fixes_) should be accompanied by one or more new automated test cases.
+
+To create a new test file, run `pnpm test:create` and follow the on-screen prompts. \
```
Comment (SirzBenjie) [inline CONTRIBUTING.md]:
```suggestion
To create a new test file, run `pnpm test:create` and follow the on-screen prompts.

```

File: CONTRIBUTING.md [hunk 19]
```diff
@@ -133,18 +132,30 @@ You can get help testing your specific changes, and you might have found a new o
...
+If the move/ability/etc. you're modifying already has tests, you can add new cases to the test file or edit existing ones.
 
-To make sure your changes didn't break any existing test cases, run `pnpm test:silent` in your terminal. You can also provide an argument to the command: to run only the Dancer (ability) tests, you could write `pnpm test:silent dancer`.
-  - __Note that passing all test cases does *not* guarantee that everything is working properly__. The project does not have complete regression testing.
+- Ensure that new test cases:
+  - Are deterministic. In other words, the test should never pass or fail when it shouldn't due to randomness. Among other things, this involves ensuring that abilities and moves are never randomly selected.
+  - Do not test multiple separate things in the same test case. If you have made two distinct changes, they should be tested in two separate cases.
+  - Cover as many edge cases as possible. A good strategy is to think of edge cases beforehand and create tests for them using `it.todo`. Once the edge case has been handled, you can remove the `todo` marker.
```
Comment (SirzBenjie) [inline CONTRIBUTING.md]:
```suggestion
Ensure that new test cases:
- Are deterministic. In other words, the test should never pass or fail when it shouldn't due to randomness. Among other things, this involves ensuring that abilities and moves are never randomly selected.
- Do not test multiple separate things in the same test case. If you have made two distinct changes, they should be tested in two separate cases.
- Cover as many edge cases as possible. A good strategy is to think of edge cases beforehand and create tests for them using `it.todo`. Once the edge case has been handled, you can remove the `todo` marker.
```
I think this reads better

File: CONTRIBUTING.md [hunk 20]
```diff
@@ -174,11 +181,19 @@ fix(move): Future Sight no longer crashes
 |_____________ Prefix
 ```
 
+> [!IMPORTANT]
+> If a save migrator, version increase or other breaking change is part of the PR, a `!` must be added before the `:`.
+
+Try to keep the PR title to 72 characters or less (GitHub cuts off commit titles longer than this).
```
Comment (SirzBenjie) [inline CONTRIBUTING.md]:
```suggestion
Try to keep the PR title to 72 characters or less, as GitHub will cut off commit titles longer than this when the PR is merged.
```

File: .github/pull_request_template.md (hunk 13 above)
Comment (Bertie690) [reply .github/pull_request_template.md]: I'm a bit wary at suggesting running `pnpm test:silent` without an argument on local runs.

Maybe also mention the CI checks - those run faster and without blocking a CPU core for 20+ mins

File: .github/pull_request_template.md (hunk 14 above)
Comment (Bertie690) [reply .github/pull_request_template.md]: and yet we still miss lots of them...

File: .github/pull_request_template.md (hunk 15 above)
Comment (Bertie690) [reply .github/pull_request_template.md:104]: I intentionally added those underscores to explicitly indicate _where_ to put the link.

I think any sane person will realize they need to delete the underscores and fill in the link lol

File: .github/pull_request_template.md (hunk 16 above)
Comment (Bertie690) [reply .github/pull_request_template.md:109]: see above

File: CONTRIBUTING.md (hunk 17 above)
Comment (Bertie690) [reply CONTRIBUTING.md]: Why remove the soft line break?
